astropy/visualization/mpl_normalize.py

```diff
@@ -212,6 +212,231 @@ def inverse(self, values, invalid=None):
+
+        if stretch == "linear":
+            stretch = LinearStretch()
+        elif stretch == "sqrt":
+            stretch = SqrtStretch()
+        elif stretch == "power":
+            stretch = PowerStretch(power)
+        elif stretch == "log":
+            stretch = LogStretch(log_a)
+        elif stretch == "asinh":
+            stretch = AsinhStretch(asinh_a)
+        elif stretch == "sinh":
+            stretch = SinhStretch(sinh_a)
+        else:
+            raise ValueError(f"Unknown stretch: {stretch}.")
+        self.stretch = stretch
+
+        self.clip = clip
+        self.invalid = invalid
+
+    def __call__(self, data):
+        """
+        Return an `ImageNormalize` instance that can be used for displaying
+        images with Matplotlib.
+
+        Parameters
+        ----------
+        data : ndarray
+            The image array.
+
+        Returns
+        -------
+        result : `ImageNormalize` instance
+            An `ImageNormalize` instance that can be used for
+            displaying images with Matplotlib.
+        """
+        vmin, vmax = self.interval.get_limits(data)
+        return ImageNormalize(
+            vmin=vmin,
+            vmax=vmax,
+            stretch=self.stretch,
+            clip=self.clip,
+            invalid=self.invalid,
+        )
+
+    def imshow(self, data, ax=None, **kwargs):
+        """
+        A convenience function to display an image using matplotlib's
+        `matplotlib.pyplot.imshow` function with the normalization
+        defined by this class.
+
+        Parameters
+        ----------
+        data : 2D or 3D array-like
+            The data to display. Can be whatever
+            `~matplotlib.pyplot.imshow` and `ImageNormalize` both
+            accept.
+
+        ax : None or `~matplotlib.axes.Axes`, optional
+            The matplotlib axes on which to plot. If `None`, then the
+            current `~matplotlib.axes.Axes` instance is used.
+
+        **kwargs : dict, optional
+            Keywords arguments passed to `~matplotlib.pyplot.imshow`.
+            Cannot include the ``norm`` or ``X`` keyword.
+
+        Returns
+        -------
+        result : `~matplotlib.image.AxesImage`
+            The `~matplotlib.image.AxesImage` generated by
+            `~matplotlib.pyplot.imshow`.
+
+        Examples
+        --------
+        .. plot::
+            :include-source:
+
+            import numpy as np
+            import matplotlib.pyplot as plt
+            from astropy.visualization import SimpleNorm
+
+            image = np.arange(65536).reshape((256, 256))
+            snorm = SimpleNorm('sqrt', percent=98)
+            fig, ax = plt.subplots()
+            axim = snorm.imshow(image, ax=ax, origin='lower')
+            fig.colorbar(axim)
+        """
+        import matplotlib.pyplot as plt
+
+        if ax is None:
+            ax = plt.gca()
+
+        if "X" in kwargs:
+            raise ValueError("Cannot give both ``X`` and ``data``")
```

Comment:
do we really need this branch ? I would expect `Axis.imshow` to handle this case itself.